    @Test
    public final void addSource() {

        try {
            root3.addSource(root2);
            Assert.fail();
        } catch (Exception e) {

        }
    }

